data-prepper-plugins/personalize-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/personalize/PersonalizeSink.java

```diff
@@ -0,0 +1,102 @@
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.sink.AbstractSink;
+import org.opensearch.dataprepper.model.sink.Sink;
+import org.opensearch.dataprepper.model.sink.SinkContext;
+import org.opensearch.dataprepper.plugins.sink.personalize.configuration.PersonalizeSinkConfiguration;
+import org.opensearch.dataprepper.plugins.sink.personalize.dataset.DatasetTypeOptions;
+import software.amazon.awssdk.services.personalizeevents.PersonalizeEventsClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Implementation class of personalize-sink plugin. It is responsible for receiving the collection of
+ * {@link Event} and uploading to amazon personalize.
+ */
+@DataPrepperPlugin(name = "aws_personalize", pluginType = Sink.class, pluginConfigurationType = PersonalizeSinkConfiguration.class)
+public class PersonalizeSink extends AbstractSink<Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PersonalizeSink.class);
+
+    private final PersonalizeSinkConfiguration personalizeSinkConfig;
+    private volatile boolean sinkInitialized;
+    private final PersonalizeSinkService personalizeSinkService;
+    private final SinkContext sinkContext;
+
+    /**
+     * @param pluginSetting dp plugin settings.
+     * @param personalizeSinkConfig personalize sink configurations.
+     * @param sinkContext sink context
+     * @param awsCredentialsSupplier aws credentials
+     * @param pluginFactory dp plugin factory.
+     */
+    @DataPrepperPluginConstructor
+    public PersonalizeSink(final PluginSetting pluginSetting,
+                           final PersonalizeSinkConfiguration personalizeSinkConfig,
+                           final PluginFactory pluginFactory,
+                           final SinkContext sinkContext,
+                           final AwsCredentialsSupplier awsCredentialsSupplier) {
+        super(pluginSetting);
+        this.personalizeSinkConfig = personalizeSinkConfig;
+        this.sinkContext = sinkContext;
+
+        sinkInitialized = false;
+
+        final PersonalizeEventsClient personalizeEventsClient = ClientFactory.createPersonalizeEventsClient(personalizeSinkConfig, awsCredentialsSupplier);
+
+        personalizeSinkService = new PersonalizeSinkService(personalizeSinkConfig, pluginMetrics);
+    }
+
+    @Override
+    public boolean isReady() {
+        return sinkInitialized;
+    }
+
+    @Override
+    public void doInitialize() {
+        try {
+            doInitializeInternal();
```

Comment:
Nit: This doesn't have to be a function if the function is just setting a boolean. Do you expect this function to do something different in future? If yes, keep it like it is now. Other wise I think it is unnecessary.

Reply:
I'll change this and add back in the future if necessary